src/foundry/client-esm/applications/api/application.d.mts

```diff
@@ -1,10 +1,38 @@
+import type { MustConform } from "../../../../types/helperTypes.d.mts";
 import type { DeepPartial, InexactPartial, MaybePromise } from "../../../../types/utils.d.mts";
 import type EventEmitterMixin from "../../../common/utils/event-emitter.d.mts";
 
 // TODO: Investigate use of DeepPartial vs Partial vs InexactPartial
 // TODO: Should
 
+declare const __Configuration: unique symbol;
+declare const __RenderOptions: unique symbol;
+declare const __RenderContext: unique symbol;
+
+type _MustBeAssignableToInternal = MustConform<ApplicationV2, ApplicationV2.Internal<any, any, any>>;
+
 declare namespace ApplicationV2 {
+  /**
+   * This type is an internal implementation detail of fvtt-types.
+   *
+   * It is used in `HandlebarsApplicationMixin` as a proxy bound for `ApplicationV2`
+   * as well as to implement generic passthrough from the mixin into the mixin
+   * class.
+   *
+   * It soundly be used as a bound to guarantee a subclass of `ApplicationV2`
+   * because it uses some `unique symbol`s that are used nowhere else in the
+   * codebase.
+   */
+  interface Internal<
```

Comment:
I think the comment here explains it well but I did this for `HandlebarsApplicationMixin`. Admittedly on a technical level there's a few other ways that could be done but I'm applying this recipe because at this point I'm concerned about the more naive patterns causing issues.